core/kits/documents/tabs/settings-layout.php

```diff
@@ -151,51 +158,98 @@ protected function register_tab_controls() {
 			]
 		);
 
-		$this->add_control(
-			'breakpoint_md_heading',
-			[
-				'label' => __( 'Mobile', 'elementor' ),
-				'type' => Controls_Manager::HEADING,
-			]
-		);
+		$prefix = Breakpoints_Manager::BREAKPOINT_OPTION_PREFIX;
+		$options = [];
 
-		$this->add_control(
-			Responsive::BREAKPOINT_OPTION_PREFIX . 'md',
-			[
-				'label' => __( 'Breakpoint', 'elementor' ) . ' (px)',
-				'type' => Controls_Manager::NUMBER,
-				'min' => $default_breakpoints['sm'] + 1,
-				'max' => $default_breakpoints['lg'] - 1,
-				'default' => $default_breakpoints['md'],
-				'placeholder' => $default_breakpoints['md'],
-				/* translators: %d: Breakpoint value */
-				'desc' => sprintf( __( 'Sets the breakpoint between tablet and mobile devices. Below this breakpoint mobile layout will appear (Default: %dpx).', 'elementor' ), $default_breakpoints['md'] ),
-			]
-		);
+		foreach ( $breakpoints as $breakpoint_key => $breakpoint ) {
+			$options[ $prefix . $breakpoint_key ] = $breakpoint->get_config( 'label' );
+		}
 
 		$this->add_control(
-			'breakpoint_lg_heading',
+			self::BREAKPOINTS_SELECT_CONTROL_ID,
 			[
-				'label' => __( 'Tablet', 'elementor' ),
-				'type' => Controls_Manager::HEADING,
+				'label' => __( 'Active Breakpoints', 'elementor' ),
+				'type' => Controls_Manager::SELECT2,
+				/* translators: 1: Mobile, 2: Tablet. */
+				'description' => __( 'Mobile and Tablet options cannot be deleted', 'elementor' ),
+				'options' => $options,
+				'default' => [
+					$prefix . $breakpoint_key_mobile,
+					$prefix . $breakpoint_key_tablet,
+				],
+				'select2options' => [
+					'allowClear' => false,
+				],
+				'nonDeletableOptions' => [
+					$prefix . $breakpoint_key_mobile,
+					$prefix . $breakpoint_key_tablet,
+				],
+				'label_block' => true,
+				'render_type' => 'none',
+				'frontend_available' => true,
+				'multiple' => true,
 			]
 		);
 
-		$this->add_control(
-			Responsive::BREAKPOINT_OPTION_PREFIX . 'lg',
-			[
+		$this->add_breakpoints_controls();
+
+		// Include the old mobile and tablet breakpoint controls as hidden for backwards compatibility.
+		$this->add_control( 'viewport_md', [ 'type' => Controls_Manager::HIDDEN ] );
+		$this->add_control( 'viewport_lg', [ 'type' => Controls_Manager::HIDDEN ] );
+
+		$this->end_controls_section();
+	}
+
+	private function add_breakpoints_controls() {
+		$breakpoints_manager = Plugin::$instance->breakpoints;
+		$breakpoints_config = $breakpoints_manager->get_config();
+		$default_breakpoints_config = Breakpoints_Manager::get_default_config();
```

Comment:
Using the static default config is necessary - when registering breakpoint controls for the kit, we cannot rely on an existing kit, to avoid an infinite loop in case the active kit is deleted and a new default kit is instantiated.

Reply:
Use the config only where it's needed. Use the instances elsewhere.

Reply:
Create `get_active_breakpoints` instead of `get_active_config`

Reply:
Done.